compiler/passes/docs.cpp

```diff
@@ -592,3 +598,103 @@ static inline std::string ltrim(std::string s) {
   s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
   return s;
 }
+
+/*
+ * Return true if 's' is empty or only has whitespace characters.
+ */
+static inline bool isEmpty(std::string s) {
+  return s.end() == std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace)));
+}
+
+/*
+ * Erase 'count' number of characters from beginning of each line in 's'. Just
+ * ltrim() the first line, though.
+ */
```

Comment:
I considered trimming trailing whitespace, too. Sphinx doesn't seem to care about it, and if someone puts it in a rst code block, it is probably intentional, in which case they would probably prefer that it not be trimmed. I decided it was easier/safer to just leave trailing whitespace in the comments. It will be trivial to add later if we decide it is useful.